### PR TITLE
[Auth, Sign] #485 remove pairing

### LINKED ISSUE #485: [Auth] Allow to remove pairing

### PR DESCRIPTION
**Pull request by llbartekll**
# Description
- Add disconnect() method to auth client api. Removes pairing
- Add disconnect service for Sign

Resolves #485 #447 
## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

File: Sources/WalletConnectSign/Engine/Common/DeletePairingService.swift [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import Foundation
+import WalletConnectKMS
+import WalletConnectUtils
+import WalletConnectPairing
+
+class DeletePairingService {
```
Comment (flypaper0) [inline Sources/WalletConnectSign/Engine/Common/DeletePairingService.swift:6]: Can we move it in WalletConnectPairing package?

Comment (flypaper0) [reply Sources/WalletConnectSign/Engine/Common/DeletePairingService.swift:6]: oh, I see. Whey using different NetworkInteractors..

Comment (llbartekll) [reply Sources/WalletConnectSign/Engine/Common/DeletePairingService.swift:6]: yeah exactly, even the protocol method is the same

File: Sources/WalletConnectSign/Engine/Common/DisconnectService.swift [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+        self.deleteSessionService = deleteSessionService
+        self.pairingStorage = pairingStorage
+        self.sessionStorage = sessionStorage
+    }
+
+    func disconnect(topic: String) async throws {
+        if pairingStorage.hasPairing(forTopic: topic) {
+            try await deletePairingService.delete(topic: topic)
+        } else if sessionStorage.hasSession(forTopic: topic) {
+            try await deleteSessionService.delete(topic: topic)
```
Comment (flypaper0) [inline Sources/WalletConnectSign/Engine/Common/DisconnectService.swift:28]: Pairing and a session at the same time? I mean: do we need to clear pairing and session together on disconnect call?

Comment (flypaper0) [reply Sources/WalletConnectSign/Engine/Common/DisconnectService.swift:28]: if so, I think is better to have 1 DeleteSessionService, which clean pairing and session storage

Comment (llbartekll) [reply Sources/WalletConnectSign/Engine/Common/DisconnectService.swift:28]: one topic corresponds to one pairing or one session and we have a common public method for deletion.
So we either delete a pairing or a session.

**Review by flypaper0 — APPROVED** [review]

**Review by flypaper0 — APPROVED** [review]

**Pull request merged**